WordPress/Classes/Services/SiteSuggestionService.swift

```diff
@@ -0,0 +1,119 @@
+        } else if ReachabilityUtils.isInternetReachable() {
+            fetchAndPersistSuggestions(for: blog, completion: completion)
+        } else {
+            completion(nil)
+        }
+    }
+
+    /**
+    Performs a REST API request for the given blog.
+    Persists response objects to Core Data.
+
+    @param blog/site to retrieve suggestions for
+    */
+    private func fetchAndPersistSuggestions(for blog: Blog, completion: @escaping ([SiteSuggestion]?) -> Void) {
+
+        // if there is already a request in place for this blog, just wait
+        guard !blogsCurrentlyBeingRequested.contains(blog) else { return }
+
+        guard let hostname = blog.hostname else { return }
+
+        let suggestPath = "/wpcom/v2/sites/\(hostname)/xposts"
+        let params = ["decode_html": true] as [String: AnyObject]
+
+        // add this blog to currently being requested list
+        blogsCurrentlyBeingRequested.append(blog)
+
+        defaultAccount()?.wordPressComRestApi.GET(suggestPath, parameters: params, success: { [weak self] responseObject, httpResponse in
+            guard let `self` = self else { return }
+
+            let context = ContextManager.shared.mainContext
+            guard let data = try? JSONSerialization.data(withJSONObject: responseObject) else { return }
+            let decoder = JSONDecoder()
+            decoder.userInfo[CodingUserInfoKey.managedObjectContext] = context
+            guard let suggestions = try? decoder.decode([SiteSuggestion].self, from: data) else { return }
+
+            // Delete any existing `SiteSuggestion` objects
+            self.retrievePersistedSuggestions(for: blog)?.forEach { suggestion in
+                context.delete(suggestion)
+            }
+
+            // Associate `SiteSuggestion` objects with blog
+            blog.siteSuggestions = Set(suggestions)
+
+            // Save the changes
+            try? ContextManager.shared.mainContext.save()
+
+            completion(suggestions)
+
+            // remove blog from the currently being requested list
+            self.blogsCurrentlyBeingRequested.removeAll { $0 == blog }
+        }, failure: { [weak self] error, _ in
+            guard let `self` = self else { return }
+
+            completion([])
+
+            // remove blog from the currently being requested list
+            self.blogsCurrentlyBeingRequested.removeAll { $0 == blog}
+
+            DDLogVerbose("[Rest API] ! \(error.localizedDescription)")
+        })
+    }
+
+    /**
+    Tells the caller if it is a good idea to show suggestions right now for a given blog/site.
+
+    @param blog blog/site to check for
+    @return BOOL Whether the caller should show suggestions
+    */
+    func shouldShowSuggestions(for blog: Blog) -> Bool {
+
+        // The device must be online or there must be already persisted suggestions
+        guard ReachabilityUtils.isInternetReachable() || retrievePersistedSuggestions(for: blog)?.isEmpty == false else {
```

Comment:
I can tested with my personal blog which (I guess) does not support cross posting and I can see the empty suggestion view. This works differently on Android. I think on Android [the checks are the same](https://github.com/wordpress-mobile/WordPress-Android/pull/13184/files#diff-665b739bf8f428f16596fdaa4ec37d9e9a43a530e69cc624b5ffbe0d2bb4bcd6R60) but there is a last step where [the results view is hidden when no results are available](https://github.com/wordpress-mobile/WordPress-Android/pull/13184/files#diff-665b739bf8f428f16596fdaa4ec37d9e9a43a530e69cc624b5ffbe0d2bb4bcd6R97).

<details>
  <summary>Screenshot</summary>

![Screen Shot 2020-12-24 at 12 06 35 PM](https://user-images.githubusercontent.com/304044/103082714-6fc4e700-45e3-11eb-9975-14f1f24d27bd.png)
</details>

Reply:
Thanks for reporting! We'll be adding pre-fetching to this iOS implementation in another PR, so I'll check then that the behavior with regard to disabling the suggestion UI in these scenarios is the same.